crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -721,7 +822,7 @@ impl CumulocityConverter {
                     let entity = self.entity_store.get(&source).unwrap();
                     if let Some(message) = external_device_registration_message(entity) {
                         self.children
-                            .insert(entity.entity_id.to_string(), Operations::default());
+                            .insert(entity.entity_id.clone().into(), Operations::default());
                         registration_messages.push(message);
```

Comment:
This are local registration messages. Don't we also to call `self.try_convert_entity_registration()` to register this entity on c8y?

- Is the expectation to have this mapper processing the local registration messages sent by itself?
- If so telemetry data might be sent to c8y before the child device registration.

Reply:
The `external_device_registration_message` should have been replaced with `try_convert_entity_registration` message as the `external_device_registration_message` is redundant and incomplete(doesn't cover services). The C8y device/service registration should happen from this scope itself, and we can't wait till the mapper processes the auto-generated registration messages on the next attempt, to avoid the telemetry message going to the cloud before its device's registration message.

Although harmless in its current form, we still need to do something to avoid reprocessing those auto-generated messages. But, as mentioned in the description, I'd like to defer that for a follow-up PR, where can handle this case, and the duplicate processing of all the retained messages(including alarms) on a mapper restart with a single mechanism, if possible. I was thinking of persisting some of the reg information on the file system to tackle this. But, that is to be discussed and I'd like to keep that out of the scope of this PR.

Reply:
> The C8y device/service registration should happen from this scope itself, and we can't wait till the mapper processes the auto-generated registration messages on the next attempt, to avoid the telemetry message going to the cloud before its device's registration message.

Correct. So do you plan to update this PR accordingly?

Reply:
Done.